datahub-web-react/src/app/ingest/source/builder/RecipeForm/RecipeForm.tsx

```diff
@@ -117,15 +118,28 @@ function RecipeForm(props: Props) {
         data?.listSecrets?.secrets.sort((secretA, secretB) => secretA.name.localeCompare(secretB.name)) || [];
     const [form] = Form.useForm();
 
+    function updateFormValue(fieldName, fieldValue) {
+        let updatedValues = YAML.parse(displayRecipe);
+        const recipeField = allFields.find((f) => f.name === fieldName);
+        if (recipeField) {
+            updatedValues = recipeField.setValueOnRecipeOverride
+                ? recipeField.setValueOnRecipeOverride(updatedValues, fieldValue)
+                : setFieldValueOnRecipe(updatedValues, fieldValue, recipeField.fieldPath);
+        }
+        const stagedRecipe = jsonToYaml(JSON.stringify(updatedValues));
+        setStagedRecipe(stagedRecipe);
+        form.setFieldsValue({ [fieldName]: fieldValue });
+    }
```

Comment:
I'm wondering if there's a clean way we could combine logic here with the logic in `updateFormValues`?